demos/link-prediction/random-walks/ctdne-link-prediction.ipynb

```diff
@@ -0,0 +1,725 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of static random walks: 1510\n"
+     ]
+    }
+   ],
+   "source": [
+    "from stellargraph.data import BiasedRandomWalk\n",
+    "\n",
+    "static_rw = BiasedRandomWalk(graph)\n",
+    "static_walks = static_rw.run(\n",
+    "    nodes=graph.nodes(), n=num_walks_per_node, length=walk_length\n",
+    ")\n",
+    "\n",
+    "print(\"Number of static random walks: {}\".format(len(static_walks)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from gensim.models import Word2Vec\n",
+    "\n",
+    "embedding_size = 128\n",
+    "temporal_model = Word2Vec(\n",
+    "    temporal_walks,\n",
+    "    size=embedding_size,\n",
+    "    window=cw_size,\n",
+    "    min_count=0,\n",
+    "    sg=1,\n",
+    "    workers=2,\n",
+    "    iter=1,\n",
+    ")\n",
+    "static_model = Word2Vec(\n",
+    "    static_walks,\n",
+    "    size=embedding_size,\n",
+    "    window=cw_size,\n",
+    "    min_count=0,\n",
+    "    sg=1,\n",
+    "    workers=2,\n",
+    "    iter=1,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For convenience, we can use the trained Word2Vec models to define helper functions that transform a node ID into a node embedding.\n",
+    "\n",
+    "NOTE: Using temporal walks means we may not be able to generate an embedding for every node in the graph, since there may not be any valid temporal walks to perform that goes through a particular node. We handle this by using zeros as embeddings for such nodes to indicate that they are uninformative."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unseen_node_embedding = np.zeros(embedding_size)\n",
+    "\n",
+    "\n",
+    "def temporal_embedding(u):\n",
+    "    try:\n",
+    "        return temporal_model.wv[u]\n",
+    "    except KeyError:\n",
+    "        return unseen_node_embedding\n",
+    "\n",
+    "\n",
+    "def static_embedding(u):\n",
+    "    return static_model.wv[u]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Node Embedding Visualisation\n",
+    "\n",
+    "For visualisation of embeddings, we'll first define a helper function that we can also use later to show the TSNE visualisation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_tsne(title, x, y=None):\n",
+    "    tsne = TSNE(n_components=2)\n",
+    "    x_t = tsne.fit_transform(x)\n",
+    "\n",
+    "    plt.figure(figsize=(7, 7))\n",
+    "    plt.title(title)\n",
+    "    if y is not None:\n",
```

Comment:
Ah thanks for providing those links! `jet` looks very monochrome-y to me.. maybe I could use the colours being used in the `plot_history` util?
```python
    colors = plt.rcParams["axes.prop_cycle"].by_key()["color"]
    color_train = colors[0]
    color_validation = colors[1]
```

Reply:
That's an option, but I don't know of a good way to specify that conveniently, without a chunk of manual code. I personally think the priority here should be obvious code, meaning we shouldn't have too much user-facing tweaking code, rather than perfect "report ready" output.

If the manual code isn't too bad, or there's another way to get better colours (like using a non-`jet` colormap), I'm all for it, though.

Reply:
I've just gone with `jet` for now, but it also meant a legend is more necessary since the colours don't match the general expectation (?) of green being true and red being false, so it does end up with slightly more code